src/components/boot/AdminUser.vue

```diff
@@ -0,0 +1,321 @@
+
+                const response = await this.$store.dispatch('server/adminUser/get', false);
+                const result = response.body;
+
+                if (response.status === 200) {
+                    this.hasUser = !!result.hasUser;
+
+                    if (this.hasUser) {
+                        this.bootState = 'success';
+                        this.bootDescription = this.$t('ui.server.adminUser.success');
+                    } else {
+                        this.bootState = 'warning';
+                        this.bootDescription = this.$t('ui.server.adminUser.warning');
+                    }
+                } else if (response.status === 501) {
+                    this.bootState = 'info';
+                    this.bootDescription = this.$t('ui.server.adminUser.unsupported');
+                } else if (response.status === 502) {
+                    this.bootState = 'info';
+                    this.bootDescription = this.$t('ui.server.adminUser.unavailable');
+                } else if (response.status === 503) {
+                    this.bootState = 'error';
+                    this.bootDescription = this.$t('ui.server.prerequisite');
+                } else {
+                    this.bootState = 'error';
+                    this.bootDescription = this.$t('ui.server.error');
+                }
+
+                this.$emit('result', 'AdminUser', this.bootState);
+            },
+
+            showAdminUser() {
+                this.$emit('view', 'AdminUser');
+            },
+
+            validate() {
+                this.error = null;
+
+                this.valid = this.$refs.username.checkValidity()
+                    && this.$refs.name.checkValidity()
+                    && this.$refs.email.checkValidity()
+                    && this.$refs.password.checkValidity();
+            },
+
+            validateEmail() {
+                this.errors.email = null;
+
+                if (this.email === '') {
+                    return;
+                }
+
+                if (!this.$refs.email.checkValidity()) {
+                    this.errors.email = this.$t('ui.server.adminUser.emailInvalid');
+                }
+            },
+
+            validatePassword() {
+                this.errors.password = null;
+
+                if (this.password === '') {
+                    return;
+                }
+
+                if (this.password.length < 8) {
+                    this.errors.password = this.$t('ui.server.adminUser.passwordLength');
+                }
+            },
+
+            async save() {
+                this.processing = true;
+
+                const response = await this.$store.dispatch('server/adminUser/set', {
+                    username: this.username,
+                    name: this.name,
+                    email: this.email,
+                    password: this.password
+                });
+
+                if (response.status === 502) {
```

Comment:
Shouldn’t we check for all failing status codes here?

```suggestion
                if (response.status !== 200) {
```

Otherwise any other 5xx status code would look like a success to the user.

Reply:
that was actually outdated, since we don't have a 502 status anymore. Failing to add a user will throw a 500 error, which will display the regular error screen of the manager.